src/main/java/org/disq_bio/disq/HtsjdkReadsRddStorage.java

```diff
@@ -11,21 +11,41 @@
 import org.disq_bio.disq.impl.file.HadoopFileSystemWrapper;
 import org.disq_bio.disq.impl.file.NioFileSystemWrapper;
 import org.disq_bio.disq.impl.formats.bam.BamSink;
-import org.disq_bio.disq.impl.formats.bam.BamSource;
 import org.disq_bio.disq.impl.formats.cram.CramSink;
-import org.disq_bio.disq.impl.formats.cram.CramSource;
 import org.disq_bio.disq.impl.formats.sam.AbstractSamSink;
 import org.disq_bio.disq.impl.formats.sam.AbstractSamSource;
 import org.disq_bio.disq.impl.formats.sam.AnySamSinkMultiple;
 import org.disq_bio.disq.impl.formats.sam.SamFormat;
 import org.disq_bio.disq.impl.formats.sam.SamSink;
-import org.disq_bio.disq.impl.formats.sam.SamSource;
 
 /** The entry point for reading or writing a {@link HtsjdkReadsRdd}. */
 public class HtsjdkReadsRddStorage {
 
   /** An option for configuring how to write a {@link HtsjdkReadsRdd}. */
-  public interface WriteOption {}
+  public interface WriteOption {
+    static AbstractSamSink getSink(
+        FormatWriteOption formatWriteOption,
+        FileCardinalityWriteOption fileCardinalityWriteOption) {
+      switch (fileCardinalityWriteOption) {
+        case SINGLE:
+          switch (formatWriteOption) {
+            case BAM:
+              return new BamSink();
+            case CRAM:
+              return new CramSink();
+            case SAM:
+              return new SamSink();
+            default:
+              throw new IllegalArgumentException("Unrecognized format: " + formatWriteOption);
+          }
+        case MULTIPLE:
+          return new AnySamSinkMultiple(SamFormat.fromFormatWriteOption(formatWriteOption));
+        default:
+          throw new IllegalArgumentException(
+              "Unrecognized cardinality: " + fileCardinalityWriteOption);
+      }
+    }
+  }
 
   /** An option for configuring which format to write a {@link HtsjdkReadsRdd} as. */
   public enum FormatWriteOption implements WriteOption {
```

Comment:
This could be implemented as (I skipped some stuff):

```java
public enum FormatWriteOption implements WriteOption {
    BAM(() -> new BamSink(), SamFormat.BAM)

    ...

    // initilalized in constructor
    private final Supplier<AbstractSamSink> supplier;
    private final SamFormat format;

    ...

    public AbstractSamSink  getSink() {
        return supplier.get();
    }

   public SamFormat getSamFormat() {
       return format;
   }
}
```